lib/seattleflu/id3c/cli/command/etl/fhir.py

```diff
@@ -553,3 +558,29 @@ def observation_resource(device: str) -> Any:
             )
         )
     }
+
+
+def create_redcap_uri(record: REDCapRecord) -> str:
+    """
+    Create a JSON data URI representing the source of a given REDCap *record*.
+
+    Used in FHIR Encounter resources as the meta.source, which ultimately winds
+    up in ID3C's ``warehouse.encounter.details`` column.
+    """
+    data_scheme = 'data:application/json'
```

Comment:
Could we have some doctests for this function? Do we ever mock a REDCap record for tests?

Reply:
Hmm, I don't think we've created REDCap records for tests. Would that be too unwieldy for a doctest?

Reply:
Mocking an `id3c.cli.redcap.Record` would be possible, but probably beyond the scope of a doctest. It could be done in a new file under _tests/_ though. But probably not a merge blocker?

Reply:
I tried to make this so much more complicated than it really is!
Our REDCAP Record object is just a dictionary with a method for getting the project_id.
For refactoring work I just created a simple test like this, just building the dictionary.

>>> parse_date_field({'field1': '2000-2-12'}, 'field1')
    datetime.datetime(2000, 2, 12, 0, 0)

Reply:
> Our REDCAP Record object is just a dictionary with a method for getting the project_id.

Yes, though in the specific case of this PR, a plain dictionary look-a-like won't suffice as the code does access the `.project` and `.id` properties.